CSharpSourceCode/Battle/BlackPowderWeapon/BlackPowderWeaponMissionLogic.cs

```diff
@@ -28,6 +28,16 @@ public FireArmsMissionLogic()
         public override void OnAgentShootMissile(Agent shooterAgent, EquipmentIndex weaponIndex, Vec3 position, Vec3 velocity, Mat3 orientation, bool hasRigidBody, int forcedMissileIndex)
         {
             var itemUsage = shooterAgent.WieldedWeapon.CurrentUsageItem.ItemUsage;
+            if (shooterAgent.WieldedWeapon.Item.Type == ItemObject.ItemTypeEnum.Thrown)
+            {
+                Mission.Missile grenade = Mission.Missiles.FirstOrDefault(m => m.ShooterAgent == shooterAgent &&
+                                                                               m.Weapon.Item.StringId.Contains("hand_grenade") &&
```

Comment:
do we really need this string comparision?

Reply:
do you mean checking of item type?